inst/www/shiny-text-input.js

```diff
@@ -0,0 +1,78 @@
+var customShinyTextInputBinding = new Shiny.InputBinding();
```

Comment:
this is very similar to the original file, with just a few changes -- that's a bad pattern. We need to avoid having this logic duplicated to avoid lots of troubles when adding functionality to this package. What can we do to extract the comon parts?

Idea: have just one js file, and pass a new attrib to define whether it is a text input or a json input. in the `initialize` method we would discover what version of input we are. Based on that we would act accordingly in other methods (with simple `if`s)